### PR TITLE
Add Zoom Functionality to OpenGL Screen

### PR DESCRIPTION
**Pull request by AdrienBousquieEPFL**
## Description
This PR introduces zoom functionality to the OpenGL screen, enabling users to dynamically zoom in and out using pinch gestures. By modifying the camera's field of view (FOV) based on user input, this enhancement improves interaction with the starmap, making exploration more intuitive and engaging.

## Implementation Details
- **Gesture Detection:**
  - Integrated Android’s `ScaleGestureDetector` to detect pinch-to-zoom gestures.
  - Mapped scale factor changes to appropriate FOV adjustments.

- **FOV Handling:**
  - Updated the `Camera` to dynamically adjust its FOV during gesture events.
  - Clamped FOV to a defined range (`10° to 90°`) to maintain usability and visual clarity.

## Tasks
This PR can be merged when :
- [x] #187 merged
- [x] #188 merged

Please ensure these are merged before proceeding with this one.

### CODE REVIEW

**Review by raniabbrk — CHANGES_REQUESTED** [review]
Great work on implementing the zoom functionality! While reviewing I left some comments for potential improvements. Here’s a quick summary:

1. Move `DEFAULT_ASPECT_RATIO` to a private property to keep the companion object focused on universal values.

2.Remove the duplicate `textureManager` initialization.

3.Re-enable the logic for disabling and re-enabling depth writing during skybox rendering. 

4. Re-include the planet in the rendering.

5. Reintroduce the `drawObjects()` method.

Another thing I thought of is adding a method to reset the FOV to its default value. I think it could make the zoom functionality even more user-friendly and could help to quickly revert the zoom level. It could be implemented like this:

  ```kotlin
  fun resetFOV() {
      fov = DEFAULT_FOV
      Matrix.perspectiveM(projMatrix, 0, fov, aspectRatio, NEAR, FAR)
  }
  ```
Let me know what you think of these ideas!

File: app/src/main/java/com/github/lookupgroup27/lookup/model/map/MapRenderer.kt [hunk 1]
```diff
@@ -91,18 +88,16 @@ class Renderer : GLSurfaceView.Renderer {
    * @param unused the GL10 interface, not used
    */
   override fun onDrawFrame(unused: GL10) {
+
+    // Clear the screen
     GLES20.glClear(GLES20.GL_COLOR_BUFFER_BIT or GLES20.GL_DEPTH_BUFFER_BIT) // Clear the screen
 
-    // Render skybox first
-    GLES20.glDepthMask(false)
```
[the remark this answers is not in the document]
Comment (AdrienBousquieEPFL) [reply app/src/main/java/com/github/lookupgroup27/lookup/model/map/MapRenderer.kt:97]: Thank you for the explanation, I will reintroduce the code.

File: app/src/main/java/com/github/lookupgroup27/lookup/model/map/MapRenderer.kt [hunk 2]
```diff
@@ -119,27 +114,6 @@ class Renderer : GLSurfaceView.Renderer {
...
-            context,
-            position,
-            color,
-            size = 1f,
-            vertexShaderCode = vertexShaderCode,
-            fragmentShaderCode = fragmentShaderCode) // Create star
-  }
-
-  /** Renders additional objects in the scene using the current MVP matrix. */
-  private fun drawObjects() {
```
[the remark this answers is not in the document]
Comment (AdrienBousquieEPFL) [reply app/src/main/java/com/github/lookupgroup27/lookup/model/map/MapRenderer.kt:138]: Thank you for clarifying the desired behavior. I apologize for any misunderstanding earlier—it wasn't entirely clear to me at the time. I'll reintroduce it as per your request.

File: app/src/main/java/com/github/lookupgroup27/lookup/model/map/Camera.kt [hunk 3]
```diff
@@ -21,16 +22,26 @@ import android.util.Log
...
+    const val MAX_FOV = 110f
+    const val DEFAULT_FOV = 45f
+    const val MIN_FOV = 1f
+
+    // Near and far clipping plane constants
     const val NEAR = 0.1f
     const val FAR = 100f
+
+    // Aspect ratio constants
+    const val DEFAULT_ASPECT_RATIO = 1f
```
[the remark this answers is not in the document]
Comment (AdrienBousquieEPFL) [reply app/src/main/java/com/github/lookupgroup27/lookup/model/map/Camera.kt:44]: That makes sense, thank you!

**Comment by AdrienBousquieEPFL** [comment]
> Another thing I thought of is adding a method to reset the FOV to its default value. I think it could make the zoom functionality even more user-friendly and could help to quickly revert the zoom level. It could be implemented like this:
> 
> ```kotlin
> fun resetFOV() {
>     fov = DEFAULT_FOV
>     Matrix.perspectiveM(projMatrix, 0, fov, aspectRatio, NEAR, FAR)
> }
> ```

Thank you for the review.
Really like the feature idea, I think we should define it more properly with the integration in the UI for the zoom. We can discuss how we could implement this feature which is a great idea.
I created it here:
- #219

Please review the issue and don't hesitate to comment or change the issue description so that the feature is in accordance with team vision.

**Review by raniabbrk — APPROVED** [review]
Thank you for implementing my suggestions, it looks good to me now. Approved for merge.

**Pull request merged**